src/content/content_location.rs

```diff
@@ -0,0 +1,114 @@
+use crate::headers::{HeaderName, HeaderValue, Headers, CONTENT_LOCATION};
+use crate::{Status, Url};
+
+use std::convert::TryInto;
+
+/// Indicates an alternate location for the returned data.
+///
+/// # Specifications
+///
+/// - [RFC 7231, section 3.1.4.2: Content-Length](https://tools.ietf.org/html/rfc7231#section-3.1.4.2)
+///
+/// # Examples
+///
+/// ```
+/// # fn main() -> http_types::Result<()> {
+/// #
+/// use http_types::{Response,Url};
+/// use http_types::content::{ContentLocation};
+///
+/// let content_location = ContentLocation::new("https://example.net/".to_string());
+///
+/// let mut res = Response::new(200);
+/// content_location.apply(&mut res);
+///
+/// let content_location = ContentLocation::from_headers(Url::parse("https://example.net/").unwrap(),res)?.unwrap();
+/// assert_eq!(content_location.location(), "https://example.net/");
+/// #
+/// # Ok(()) }
+/// ```
+#[derive(Debug)]
+pub struct ContentLocation {
+    url: Url,
+}
+
+impl ContentLocation {
+    /// Create a new instance of `Content-Location` header.
+    pub fn new(url: String) -> Self {
+        Self { url : Url::parse(&url).unwrap() }
+    }
+
+    /// Create a new instance from headers.
+    pub fn from_headers<U>(base_url: U, headers: impl AsRef<Headers>) -> crate::Result<Option<Self>>
+    where
+        U: TryInto<Url>,
+        U::Error: std::fmt::Debug,
+   {
+        let headers = match headers.as_ref().get(CONTENT_LOCATION) {
+            Some(headers) => headers,
+            None => return Ok(None),
+        };
+
+        // If we successfully parsed the header then there's always at least one
+        // entry. We want the last entry.
+        let value = headers.iter().last().unwrap();
+        let base = base_url.try_into().expect("Could not convert into a valid url");
```

Comment:
```suggestion
        let base = base_url.try_into()?;
```
(Let's not crash a server on foreign input.)

Reply:
Hi @Fishrock123, thanks for the feedback! yes, that sound good 👍 
Thx!